development/webpack/build.ts

```diff
@@ -0,0 +1,62 @@
+import { webpack } from 'webpack';
+import type WebpackDevServerType from 'webpack-dev-server';
+import { noop, logStats, __HMR_READY__ } from './utils/helpers';
+import config from './webpack.config.js';
+
+// disable browserslist stats as it needlessly traverses the filesystem multiple
+// times looking for a stats file that doesn't exist.
+require('browserslist/node').getStat = noop;
+
+/**
+ * Builds the extension
+ *
+ * @param onComplete
+ */
+export function build(onComplete: () => void = noop) {
+  const isDevelopment = config.mode === 'development';
+
+  const { watch, ...options } = config;
+  const compiler = webpack(options);
+  if (__HMR_READY__ && watch) {
```

Comment:
HMR can't be ready until our circular dependencies are gone. This code block is mostly untested and problem doesn't work.